File: CS50X/sentimental-readability/readability.py
from cs50 import get_string
from curses.ascii import isalpha


def main():

    text = get_string('Text: ')
    letters_no = int(0)
    # Count letter
    for i in range(len(text)):
        if isalpha(text[i]):
            letters_no += 1
    # Count blanks
    blank_no = int(0)
    for i in range(len(text)):
        if text[i] == ' ':
            blank_no += 1
    # Count words
    word_no = blank_no + 1
    # Count . ! ?
    sentence_no = int(0)
    for i in range(len(text)):
        if text[i] == '.':
            sentence_no += 1
        if text[i] == '!':
            sentence_no += 1
        if text[i] == '?':
            sentence_no += 1
    # Compute Coleman-Liau index
    L = letters_no / word_no * 100
    S = sentence_no / word_no * 100
    index = (0.0588 * L) - (0.296 * S) - 15.8
    score = round(index)
    # Print results
    if (score < 1):
        print('Before Grade 1')
    elif (score >= 16):
        print('Grade 16+')
    else:
        for i in range(2, 16):
            if score == i:
                print('Grade ', i)


if __name__ == "__main__":
    main()

File: CS50X/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get text from user
    string text = get_string("Text: ");
    int l_no = count_letters(text);
    int w_no = count_words(text);
    int s_no = count_sentences(text);
    // Compute Coleman-Liau index
    float L = (float)l_no / w_no * 100;
    float S = (float)s_no / w_no * 100;
    float index = (0.0588 * L) - (0.296 * S) - 15.8;
    int score = round(index);
    // Print results
    if (score < 1)
    {
        printf("Before Grade 1");
    }
    else if (score >= 16)
    {
        printf("Grade 16+");
    }
    else
    {
        for (int i = 2; i < 16 ; i++)
        {
            if (score == i)
            {
                printf("Grade %i", i);
            }
        }
    }
    printf("\n");
}
// Function that counts letter
int count_letters(string text)
{
    int l_no = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            l_no++;
        }
    }
    return l_no;
}
// Function to counts words
int count_words(string text)
{
    int w_no = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isblank(text[i]))
        {
            w_no++;
        }
    }
    return w_no;
}
// Function that counts sentences
int count_sentences(string text)
{
    int s_no = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            s_no++;
        }
    }
    return s_no;
}